app/uk/gov/hmrc/gform/core/parsers/BasicParsers.scala

```diff
@@ -52,28 +55,70 @@ object BasicParsers {
 
   implicit val W = Whitespace(() | """\s+""".r)
 
-  def nextOrPrevious[A](string: String, fn: (Int, Int) => A): Parser[A] =
+  def nextOrPreviousValue[A](string: String, fn: (Int, Int) => A): Parser[A] =
+    (string ~ exactMonthDay ^^ { (loc, _, month, day) =>
+      fn(month, day)
+    })
+
+  def nextOrPrevious[A](string: String, fn: (Month, Day) => A): Parser[A] =
```

Comment:
want to see tests where this is interrogated for last day of the year going next or first day of year going previous or appropriate